sorc/IMSaggregate_mod.f90

```diff
@@ -153,20 +159,62 @@ subroutine calculate_scfIMS(idim, jdim, otype, yyyymmddhh, jdate, IMS_obs_path,
                     enddo 
                   enddo 
                 enddo
-
             else 
                 print *, 'unknown lsm:', lsm, ', choose 1 - noah, 2 - noah-mp' 
                 stop 10 
             endif
         else 
                 sndIMS=nodata_real
         endif ! skip_SD
+        ! add station identification with txxxxyyyy string
+         do t=1,6
+           do i=1,idim
+             do j=1,jdim
+                write(tile_str, '(i1)') t ! assuming <10 tiles.
+        ! create xindex string for final station identification
+                if(i<=9) then
+                  write(istr1,'(i1)') i
+                  xind='000'//istr1
+                else if (i >= 10 .AND. i <= 99) then
+                  write(istr2,'(i2)') i
+                  xind='00'//istr2
+                else if (i >= 100 .AND. i <= 999) then
+                  write(istr3,'(i3)') i
+                  xind='0'//istr3
+                else
+                  write(istr4,'(i4)') i
+                  xind=istr4
+                end if
```

Comment:
can't you specify leading zeros in the write statement format?

Reply:
`'(I4.4)'`

Reply:
Corrected as suggested. I test both nc and ascii input. Thank you for simplifying the code.